articles/130_ros_time.md

```diff
@@ -39,85 +39,89 @@ Slower than real time simulation is necessary for complicated systems where accu
 
-##### No Advanced Estimating Clock
+Users can also add callbacks to `Node::time_reversing` and `Node::time_jumping_back`.
+The developer has the opportunity to register callbacks with the handler to clear any state from their system if necessary before time will be in the past.
+If the time on the clock jumps backwards, a callback handler will be invoked and be required to complete before any calls to the ROS time abstraction report the new time.
 
-There are more advanced techniques which could be included to attempt to estimate the propagation properties and extrapolate between time ticks.
-However all of these techniques will require making assumptions about the future behavior of the time abstraction.
-And in the case that playback or simulation is instantaneously paused, it will break any of these assumptions.
-There are techniques which would allow potential interpolation, however to make these possible it would require providing guarantees about the continuity of time into the future.
-For more accuracy the progress of time can be slowed, or the frequency of publishing can be increased.
-Tuning the parameters for the `/clock` topic lets you trade off time for computational effort and/or bandwidth.
+The frequency of publishing the `/clock` as well as the granularity are not specified as they are application specific. Assuming they latch one message, they shouldn't need to publish more often than the multiplier changes. If the realtime multiplier cannot be computed, the message will need to be published very frequently. 
 
 #### Custom Time Source
 
-It is possible that the user may have access to an out of band time source which can provide better performance than the default source the `/clock` topic.
+It is possible that the user may have access to an out-of-band-time source which can provide better performance than the default source the `/clock` topic.
 It might be possible that for their use case a more advanced algorithm would be needed to propagate the simulated time with adequate precision or latency with restricted bandwidth or connectivity.
 The user will be able to switch out the time source for either each instance of their Time object as well as have the ability to override the default for the process.
+This can be acheived in C++ using the template on the Time object. The RCL package also contains a method to disable `ROSTime`.
 
-It is possible to use an external time source such as GPS in as a ROSTime source, but it is recommended to integrate a time source like that using standard ntp integrations with the system clock since that is already an established mechanism and will not need to deal with more complicated changes such as time jumps.
-
-## Implementation
+It is possible to use an external time source such as GPS as a ROSTime source. When writing this publisher, ensure that you handle jumps in time (both forward and back) and set the realtime multipleir correctly in those situations.
```

Comment:
Also, I feel like this is an abuse of ROS Time. Why not integrate those sources into the system time? This is the same argument for how to integrate PTP (IEEE 1588) or something like TICSync (http://ieeexplore.ieee.org/document/5980112/?reload=true).

I'm curious to know why you would want to have a separate ROS specific mechanism for using those sources rather than just integrating those into the system time of the machine.